LoRaEngine/test/LoRaWan.IntegrationTest/C2DMessageTest.cs

```diff
@@ -30,6 +35,19 @@ public C2DMessageTest(IntegrationTestFixtureCi testFixture)
         {
         }
 
+        /// <summary>
+        /// Ensures that a cloud to device message has not been seen more than expected
+        /// </summary>
+        /// <param name="foundCount"></param>
+        private void EnsureNotSeenToManyTimes(int foundCount)
+        {
+            Assert.True(foundCount <= CloudToDeviceMessageReceiveCountThreshold, $"Cloud to device message was processed {foundCount} times");
+            if (foundCount > 1)
+            {
+                TestLogger.Log($"[WARN] Cloud to device message was processed {foundCount} times");
```

Comment:
I would suggest consistently using "C2D message" in all logging.

Reply:
We changed all errors to be explicitly say "cloud to device message" as c2d is a lingo.